x-pack/solutions/observability/plugins/infra/public/pages/logs/page_content.tsx

```diff
@@ -77,9 +81,35 @@ export const LogsPageContent: React.FunctionComponent = () => {
         <Route
           path="/stream"
           exact
-          render={() => {
-            share.url.locators.get<LogsLocatorParams>(LOGS_LOCATOR_ID)?.navigate({});
-
+          render={({ location }) => {
+            const searchParams = new URLSearchParams(location.search);
+            const g = searchParams.get('_g') || '';
+            const a = searchParams.get('_a') || '';
+
+            const gState = (safeDecode(g) || {}) as unknown as DiscoverGlobalState;
+            const aState = (safeDecode(a) || {}) as unknown as DiscoverAppState;
+
+            const dataViewId =
+              aState.dataSource?.type === 'dataView'
+                ? (aState.dataSource as DataViewDataSource).dataViewId
+                : '';
+
+            const locatorParams = {
+              dataViewId,
+              timeRange: gState.time || { from: 'now-15d', to: 'now' },
+              filters: [
+                ...(Array.isArray(gState.filters) ? gState.filters : []),
+                ...(Array.isArray(aState.filters) ? aState.filters : []),
+              ] as Filter[],
+              query:
+                aState.query && typeof aState.query === 'object'
+                  ? aState.query
+                  : { language: 'kuery', query: '' },
+              columns: Array.isArray(aState.columns) ? aState.columns : [],
+              sort: Array.isArray(aState.sort) ? aState.sort : [['@timestamp', 'desc']],
+            };
+
+            share.url.locators.get(DISCOVER_APP_LOCATOR)?.navigate(locatorParams);
```

Comment:
```suggestion
            share.url.locators.get<DiscoverAppLocatorParams>(DISCOVER_APP_LOCATOR)?.navigate(locatorParams);
```
Nit: If nothing else, I'd recommend typing the locator when retrieving it here.